### PR TITLE
DAOS-11396 control: Add NUMA affinity to tmpfs mount options

### PR DESCRIPTION
**Pull request by mjmac**
Use the mpol:prefer=Node option to prefer memory allocation
from the NUMA node used by the engine.

Required-githooks: true

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]

File: src/control/server/config/server_test.go [hunk 1]
```diff
@@ -1446,9 +1444,9 @@ func TestConfig_SetEngineAffinities(t *testing.T) {
 					engine.MockConfig().
 						WithFabricInterface("ib0").
 						WithFabricProvider("ofi+verbs").
-						WithPinnedNumaNode(1),
+						WithPinnedNumaNode(0),
 				),
-			expNumaSet: []int{1},
+			expNumaSet: []int{0},
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/server/config/server_test.go:1449]: Good catch, I meant to revert that before submitting. I was experimenting with some stuff while trying to get things working.

File: src/control/server/config/server_test.go [hunk 2]
```diff
@@ -1375,10 +1376,7 @@ func TestConfig_setEngineAffinity(t *testing.T) {
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
-			log, buf := logging.NewTestLogger(t.Name())
-			defer ShowBufferOnFailure(t, buf)
-
-			err := setEngineAffinity(log, tc.cfg, tc.setNUMA)
+			err := tc.cfg.SetNUMAAffinity(tc.setNUMA)
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/server/config/server_test.go:1379]: Done.

File: src/control/server/storage/scm.go [hunk 3]
```diff
@@ -303,7 +303,8 @@ type (
 	// RamdiskParams defines the sub-parameters of a Format operation that
 	// will use tmpfs-based ramdisk
 	RamdiskParams struct {
-		Size uint
+		Size     uint
+		MpolNode uint // NUMA node to bind ramdisk to
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/server/storage/scm.go:307]: Yeah, I was thinking that the names here should correspond to the parameters used to mount a ramdisk, but I can see the argument being made that this field value could eventually be used for more than one parameter.

**Review by tanabarr — APPROVED** [review]

**Comment by mjmac** [comment]
Two failures due to the logfile issue, fine otherwise.

**Pull request merged**